payjoin-cli/src/db/v2.rs

```diff
@@ -1,21 +1,54 @@
+use std::sync::Arc;
+
 use bitcoincore_rpc::jsonrpc::serde_json;
+use payjoin::persist::{PersistableValue, Persister};
 use payjoin::receive::v2::Receiver;
 use payjoin::send::v2::Sender;
-use sled::{IVec, Tree};
+use sled::Tree;
 use url::Url;
 
 use super::*;
 
-impl Database {
-    pub(crate) fn insert_recv_session(&self, session: Receiver) -> Result<()> {
-        let recv_tree = self.0.open_tree("recv_sessions")?;
-        let key = &session.id();
-        let value = serde_json::to_string(&session).map_err(Error::Serialize)?;
-        recv_tree.insert(key.as_slice(), IVec::from(value.as_str()))?;
+#[derive(Clone)]
+pub(crate) struct SenderPersister(pub Arc<Database>);
```

Comment:
Would make more sense to have a constructor than pub Arc<Database>

Reply:
Yeah agreed. Fixed